src/rng.rs

```diff
@@ -47,12 +47,20 @@ use core::{mem, slice};
 /// use rand::Rng;
 ///
 /// fn foo<R: Rng + ?Sized>(rng: &mut R) -> f32 {
-///     rng.gen()
+///     rng.random()
 /// }
 ///
 /// # let v = foo(&mut thread_rng());
 /// ```
 pub trait Rng: RngCore {
+    /// Alias for [`Rng::random`].
+    #[inline]
+    #[deprecated(since="0.9.0", note="This method conflicts with a keyword in Rust 2024. Use `Rng::random` instead.")]
+    fn gen<T>(&mut self) -> T
+    where Standard: Distribution<T> {
+        self.random()
+    }
+
```

Comment:
This should probably go at the bottom of the impl block so that `random` shows up first in documentation.

Reply:
We can just use `#[doc(hidden)]` instead.

I suggest a shorter message:

> renamed to `random` to avoid conflict with the new `gen` keyword in Rust 2024

Reply:
Hiding it isn't a good idea. There's countless uses of `gen` in existing code and tutorials, so people should still be able to find it and read its documentation in rustdoc and rust-analyzer.

Reply:
Why do we want to hide it? It will already be marked as deprecated in the docs, won't it?

Reply:
To reduce clutter... but keeping it visible is also acceptable.